src/main/java/com/connexta/poller/service/StatusServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.poller.service;
+
+import static com.dyngr.core.AttemptResults.continueFor;
+import static com.dyngr.core.AttemptResults.justContinue;
+import static com.dyngr.core.AttemptResults.justFinish;
+
+import com.connexta.store.exceptions.DetailedResponseStatusException;
+import com.dyngr.PollerBuilder;
+import com.dyngr.core.AttemptMaker;
+import com.dyngr.core.StopStrategies;
+import com.dyngr.core.WaitStrategies;
+import java.net.URI;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+/**
+ * The class's responsibilities are to manage the settings for polling and kick off new polling
+ * tasks. Other responsibilities include settings the wait and retry behavior.
+ */
+@Slf4j
+@AllArgsConstructor
+public class StatusServiceImpl implements StatusService {
+
+  @Min(0)
+  private final int secondsBetweenRetries;
+
+  @Min(1)
+  private final int secondsToLive;
+
+  @NotNull private final ExecutorService executorService;
+  @NotNull private final WebClient webClient;
+
+  @Override
+  @SuppressWarnings("unused")
+  public void submit(URI uri) {
+    builder().polling(getAtemptMaker(uri)).build().start();
+  }
+
+  private AttemptMaker<Void> getAtemptMaker(URI uri) {
+    return () -> {
+      StatusResponse statusResponse = null;
+      try {
+        statusResponse =
+            webClient
+                .get()
+                .uri(uri)
+                .retrieve()
+                .onStatus(
+                    httpStatus -> !HttpStatus.OK.equals(httpStatus),
+                    clientResponse ->
+                        Mono.error(
+                            () ->
+                                new DetailedResponseStatusException(
+                                    clientResponse.statusCode(),
+                                    String.format("Could not get status for %s", uri.toString()))))
+                .bodyToMono(StatusResponse.class)
+                .block();
+      } catch (Exception e) {
+        continueFor(e);
+      }
+
+      // TODO: Follow-on ticket -- If the transform job is done, stop polling
+      if (statusResponse != null && "complete".equalsIgnoreCase(statusResponse.getStatus())) {
+
+        // TODO: Is this were the everyone expects the post-status logic to happen?
+
+        // Stop polling
+        return justFinish();
+      }
+
+      // Job not done, keep polling
+      return justContinue();
+    };
+  }
+
+  private PollerBuilder builder() {
+    return PollerBuilder.newBuilder()
+        .withExecutorService(executorService)
```

Comment:
❓ Do we really want to continue if an exception happens?

Reply:
Great question! :grey_question: The answer is yes. If the transform service is down, Spring throws an exception when we make the HTTP request. Also, if polling starts before the status endpoint knows about the job, the endpoint will return a 404 and Spring will turn that response into an exception. Because of situations like that, the application should continue to poll status endpoint because the desired resource might become available. The unit test `testTemporaryFailure` does not succeed is this value is set to `true`.